File: 프로그래머스/unrated/181904. 세로 읽기/세로 읽기.java
//1. 기본 풀이. charAt()메서드를 호출하는 것 보다. index로 접근하는게 더 효율적

class Solution {
    public String solution(String my_string, int m, int c) {
        if(m==1) return my_string;
        StringBuilder answer = new StringBuilder();
        char[] chs = my_string.toCharArray();
        for(int i = c-1; i < chs.length; i += m){
            answer.append(chs[i]);
        }
        return answer.toString();
    }
}

File: 프로그래머스/0/181904. 세로 읽기/세로 읽기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string my_string, int m, int c) {
    string answer = "";
    vector<string> strArr;
    for(int i = 0; i < my_string.size(); i += m){
        strArr.push_back(my_string.substr(i, m));
    }
    for(string str : strArr){
        answer += str[c-1];
    }
    return answer;
}